dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -329,9 +329,7 @@ private void doReceived(Response res) {
         lock.lock();
         try {
             response = res;
-            if (done != null) {
-                done.signal();
-            }
+            done.signalAll();
```

Comment:
Hi, this change is irrelevant.  And is addressed in #3681 . Please rebase with the master and send the pull request again.

Reply:
OK，I will send the pull request again.